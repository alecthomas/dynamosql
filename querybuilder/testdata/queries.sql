SELECT * FROM gamescores WHERE UserId = :UserId
SELECT * FROM gamescores WHERE UserId = :UserId AND TopScore > :MinTopScore
SELECT * FROM gamescores WHERE UserId = :UserId AND GameTitle BETWEEN :MinGameTitle AND :MaxGameTitle AND TopScore > :MinTopScore
SELECT * FROM gamescores WHERE UserId = "103" AND GameTitle BETWEEN "Galaxy" AND "Meteor" AND TopScore > 1000
SELECT * FROM gamescores WHERE UserId = "103" AND begins_with(GameTitle, "Galaxy")
SELECT UserId, TopScore, Scores[3], Scores[3][2], Studio.Name, Studio.Location.Country, Studio.Employees[3] FROM gamescores WHERE UserId = :UserId
