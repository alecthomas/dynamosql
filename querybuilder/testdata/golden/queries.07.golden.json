{
  "Query": "SELECT title, year FROM movies WHERE title = :title AND year > 2009 AND escaped = TRUE",
  "Prepared": {
    "Query": {
      "AttributesToGet": null,
      "ConditionalOperator": null,
      "ConsistentRead": null,
      "ExclusiveStartKey": null,
      "ExpressionAttributeNames": {
        "#escaped": "escaped",
        "#year": "year"
      },
      "ExpressionAttributeValues": null,
      "FilterExpression": "#escaped = :_gen2",
      "IndexName": null,
      "KeyConditionExpression": "title = :title AND #year > :_gen1",
      "KeyConditions": null,
      "Limit": null,
      "ProjectionExpression": "title, #year",
      "QueryFilter": null,
      "ReturnConsumedCapacity": null,
      "ScanIndexForward": null,
      "Select": null,
      "TableName": "movies"
    },
    "Columns": [
      {
        "Function": null,
        "DocumentPath": {
          "Fragment": [
            {
              "Symbol": "title",
              "Indexes": null
            }
          ]
        }
      },
      {
        "Function": null,
        "DocumentPath": {
          "Fragment": [
            {
              "Symbol": "year",
              "Indexes": null
            }
          ]
        }
      }
    ],
    "FreeParams": {
      ":title": {}
    },
    "FixedParams": {
      ":_gen1": 2009,
      ":_gen2": true
    }
  }
}
